include/koa/authentication/passport-strategies/salesforce_authentication_strategy.js

```diff
@@ -0,0 +1,65 @@
+const CustomStrategy = require('passport-custom').Strategy;
+const strategyServices = require('./strategy_services');
+const request = require('request-promise');
+const Promise = require('bluebird');
+const SALESFORCE_OAUTH_TOKEN_URL = process.env.SALESFORCE_OAUTH_TOKEN_URL || '';
+const SALESFORCE_OAUTH_AUTHORIZE_URL = process.env.SALESFORCE_OAUTH_AUTHORIZE_URL || '';
+
+module.exports = (pb) => {
+    async function getSalesforceSettings(req) {
+        let pluginService = new pb.PluginService({
+            site: req.ctx.__site
+        });
+        pluginService = Promise.promisifyAll(pluginService);
+        const settings = await pluginService.getSettingsKVAsync('tn_join');
+        return settings;
+    }
+
+    async function salesforceCallback(req, done) {
+        const loginContext = req.session._loginContext || {};
+        const state = req.url.query.state;
+        const code = req.url.query.code;
+        const settings = await getSalesforceSettings(req);
+        const options = {
+            url: `${SALESFORCE_OAUTH_TOKEN_URL}?client_id=${settings.salesforce_client_id}&redirect_uri=${settings.salesforce_callback_url}&grant_type=authorization_code&code=${code}&client_secret=${settings.salesforce_client_secret}`,
+            method: 'POST',
+            json: true
+        };
+        const response = await request(options);
+        const accessToken = response.access_token;
+        let salesforceUser = await request.get(`https://login.salesforce.com/services/oauth2/userinfo?format=json&access_token=${accessToken}`);
```

Comment:
where are you handling or reporting the errors?  you should add a try catch block here

Reply:
Just added error handling